blog-post/blog-post.md

```diff
@@ -0,0 +1,563 @@
+syntax. After quite some internal discussion and considering user feedback, the current feeling
+within the team is that we would prefer to prioritize smoothing the transition over introducing
+(subjectively) cleaner syntax.
+
+With that said, some degree of breakage may still be required. Right now we are considering the
+following changes:
+
+- Postfix types: Supporting the kind of rich types that Core enables with a prefix syntax is quite
+challenging from a parser implementation point of view. Certain use cases may not be possible at
+all, and those that are may require complex lookahead or backtracking logic. A final decision here
+will not be made until we have completed a deeper analysis of the tradeoffs.
+- Ternary operator: Changes may be required here to resolve ambiguities in the language grammar.
+- Function type / parameter syntax: The introduction of higher order functions and the more
+functional flavour of Core lends itself much more to the usage of function parameters. While likely not
+strictly necessary for parsing / grammar reasons, we suspect that a simpler syntax here will likely
+be worth the change given the benefits to readability and usability.
+
+### Semantic Breakage and Feature Removal
+
+By far the largest breakage will be the removal of inheritance. Our strong belief is that traits and
+generics are a more flexible and powerful system for polymorphism and code sharing, and that
+supporting both approaches will not be in the best interests of either the users or developers of
+Solidity over the long term. We operate in a high assurance domain where simplicity of semantics and
+ease of audit are critical. Keeping the language as simple and focused as possible has deep long
+term benefits for ecosystem security.
+
+In addition we are considering changes to some core language features that we currently consider
+problematic or unsatisfying, in particular: Libraries, Try/Catch, and Function Pointers. Input or
+feedback on the ways in which these features do or do not work for you is interesting and very
+welcome. We will publish more on this area once we have concrete plans and implementations to share.
+
+
+## What's next?
+
+The evolution of Solidity has reached a crucial moment. While the language's
+established features have successfully powered the ecosystem, certain patterns have
+revealed limitations in scalability, security, and clarity. This blog post outlines
+Argot Collective's vision for Core Solidity, a deliberate re-design of the language's
+foundation. This initiative aims to replace legacy mechanisms with more robust,
+composable, and community-driven primitives, ensuring the language remains a secure
+and efficient basis for the next generation of smart contracts.
+
+Our next steps will involve:
+
+- Type aliases and definitions: Currently the Core Solidity prototype do not support
+  the definition of type aliases and structs.
```

Comment:
This bit mixes the things that are "next" for the prototype with things that will change in the language. I think we should make a clear distinction between the two.

IMO we should focus on the language here and put stuff about solcore in `Current status`.

Structs are already available in Classic so adding them is only a change for the prototype, not for the language. On the other hand Classic does not have type aliases, so that is relevant here. We might want to make the distinction between aliases and the types defined with `data` though.

Reply:
In the same vein, there's this bit in the intro:

> but we want to add at least compile time evaluation and modules.

We already have modules in Classic Solidity and the goal is more to bring solcore to parity with them than invent something new. So I'd classify that as a prototype change.

Reply:
What might be better to focus on instead is some kind of namespacing mechanism other than modules. I'm not sure if that being tied to modules is the best design choice. In Classic that need is filled by libraries. If they get removed, the lack of more fine-grained namespacing leaves a bit of a hole.

Another thing to consider are exports: https://github.com/argotorg/solidity/issues/6482

Reply:
Exports and namespacing are both features that I think we should support. I agree that namespacing is to some degree orthogonal to modules. I also suspect that the right place to end up design wise is something that is broadly very similar to the current module system, potentially with a few tweaks and upgrades to allow for implementation hiding and stronger abstraction boundaries.

I'm not sure I understand your problem with the current text tbh. Would it help to state that compatibility with the classic module system is a goal? Or that we do not expect a broad divergence from the classic system? The new language simply does not yet have a module system, and we need one before we can consider the typesystem stable, so I think the statement that we intend to add one is correct?